packages/block-library/src/query/utils.js

```diff
@@ -89,3 +89,23 @@ export const usePostTypes = () => {
 	);
 	return { postTypesTaxonomiesMap, postTypesSelectOptions };
 };
+
+/**
+ * Recurses over a list of blocks and returns the first found
+ * Query Loop block's clientId.
+ *
+ * @param {WPBlock[]} blocks The list of blocks to look through.
+ * @return {string} The first found Query Loop's clientId.
```

Comment:
In case the search fails:

```suggestion
 * @return {string=} The first found Query Loop's clientId.
```